scripts/zones/Promyvion-Vahzl/mobs/Solicitor.lua

```diff
@@ -1,10 +1,34 @@
 -----------------------------------
 -- Area: Promyvion - Vahzl
 --   NM: Solicitor
+-- TODO: Verify cmbDelay
+-----------------------------------
+mixins = {require("scripts/mixins/families/empty_terroanima")}
 -----------------------------------
 local entity = {}
 
-entity.onMobDeath = function(mob, player, isKiller)
+entity.onMobSpawn = function(mob)
+    mob:setMod(xi.mod.DOUBLE_ATTACK, 20)
+end
+
+entity.onMobWeaponSkillPrepare = function(mob, target)
+    local roll = math.random()
+    if mob:getHPP() <= 35 then
+        if roll <= 0.7 then
+            return 1234 -- Carousel
+        else
+            return 1274 -- Impalement
+        end
+    end
+end
+
+entity.onMobFight = function(mob, target)
+    if mob:getTP() >= 2000 then
```

Comment:
I'm not sure the purpose of this, the normal AI will make it go when it has TP based on its HP input. On retail if something going frequently that typically means its gaining the TP to do it sooner or was reaction triggered so it didn't need any.